docs/proposals/202106120-edgex-integration.md

```diff
@@ -0,0 +1,144 @@
+- Required deployments : For the deployments in this group are mandatory and will be deployed with each EdgeX instance.
+- Optional deployments: For deployments in this group is optional can user can specify which deployments they want to deploy.
+
+Consider it is quite common to have multiple EdgeX instance in a single Kubernetes Cluster, With traditional service+deployment solutions, for the components in each EdgeX instance it require a unique name. When the EdgeX instance number increased the name management will be a critical issue need to address. In this proposal we leverage the United Deployment feature of OpenYurt. In that way all the EdgeX Instance share the same service name which can avoid the service name explosion.
+
+![Deployment Topology](../img/edgex/overview.png)
+
+As the deployment topology diagram shows.  Each components in the EdgeX instance will have a corresponding service mapping to it. The service will mapping to a United Deployment instance(In the backend it is actually mapping to all the deployments inside the united deployment) On the edge side a deployment with a unique name will be created and registered in the service's endpoints. Service topology policy will be automatically applied by United Deployment controller to make sure the traffic will be limited to Edge/Node Pool level.From one side we can make the management easier and on the other side we can still keep the EdgeX instance isolation. 
+#### 1. EdgeX Customer Resource Definition
+
+Following is the definition of `EdgeX` CRD and its related Golang struct. The CRD will be created when the EdgeX controller is up and running. 
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    ...
+  creationTimestamp: null
+  name: edgex.apps.openyurt.io
+spec:
+  ...
+  optionalComponents:
+    description: 
+    type: object
+
+```
+
+
+```go
+type EdgeX struct {
+    //TODO
+}
+
+```
+When user create an EdgeX CR, the EdgeX version field is required. For each specific version, the required components section are fixed and will be created automatically by the EdgeX controller. If user want run any optional components or components created by themselves, those components must be declared explicitly in the optionalComponents section. 
+
+![EdgeX Instance](../img/edgex/edgexinstance.png)
+As the EdgeX instance diagram shows, the following 8 Components are required for EdgeX Hanoi version
+- Core Data service: The core data micro service provides centralized persistence for data collected by devices.
```

Comment:
Device Service components are forgotten?

Reply:
Device service is optional since different usage scenario will require different device service. But you reminder me that it should include the device controller :) as a required component.

Reply:
Device controller is yurt-device-controller?  if Device controller is `yurt-device-controller`,  we should not deploy `yurt-device-controller` with EdgeX components because some other iot system maybe integrated into `yurt-device-controller` in the future.